File: src/test/ruby/lz77test.rb
require 'test/unit'
require 'temp/lz77'

class LZ77Test < Test::Unit::TestCase
	def setup
    	@encodedTextData = "ababassbabasbabbba` '&bs` 1 sb` 4!` . abaaaa` *!"
		@textData = "ababassbabasbabbbaababassbababsbasbasbbabbbababababaaaaabbbab"
		@compressor = Compressor.new
	end

	def testCompress
		result = @compressor.compress(@textData)
		assert_equal(@encodedTextData, result)
	end
	
	def testDecompress
		result = @compressor.decompress(@encodedTextData)
		assert_equal(@textData, result)
	end
end

File: src/test/php/lz77Test.php
<?php
require_once 'PHPUnit/Framework.php';
require_once 'main/php/lz77.php';

class LZ77Test extends PHPUnit_Framework_TestCase {

	protected $compressor;
	protected $textData = "ababassbabasbabbbaababassbababsbasbasbbabbbababababaaaaabbbab";
	protected $encodedTextData = "ababassbabasbabbba` '&bs` 1 sb` 4!` . abaaaa` *!";

	protected function setUp() {
		$this->compressor = new LZ77();
	}

	public function testCompress() {
		$this->assertNotNull($this->compressor);
		$result = $this->compressor->compress($this->textData);
		$this->assertNotNull($result);
		$this->assertEquals($this->encodedTextData, $result);
	}

	public function testDecompress() {
		$this->assertNotNull($this->compressor);
		$result = $this->compressor->decompress($this->encodedTextData);
		$this->assertNotNull($result);
		$this->assertEquals($this->textData, $result);
	}
}

?>
